src/execution_tree/primitives/sum_operation.cpp

```diff
@@ -0,0 +1,290 @@
+        std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {}
+
+    primitive_argument_type sum_operation::sum0d(arg_type&& arg,
+        hpx::util::optional<std::int64_t> axis, bool keep_dims) const
+    {
+        if (axis && axis.value() != 0 && axis.value() != -1)
+        {
+            /*HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "sum_operation::sum0d",
+                execution_tree::generate_error_message(
+                    "the sum_operation primitive requires operand axis to be "
+                    "either 0 or -1 for scalar values. " +
+                        std::to_string(axis.value()) " was provided.",
+                    name_, codename_));*/
+        }
+
+        return primitive_argument_type{arg.scalar()};
+    }
+
+    primitive_argument_type sum_operation::sum1d(arg_type&& arg,
+        hpx::util::optional<std::int64_t> axis, bool keep_dims) const
+    {
+        if (axis && axis.value() != 0 && axis.value() != -1)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "sum_operation::sum1d",
+                execution_tree::generate_error_message(
+                    "the sum_operation primitive requires operand axis to be "
+                    "either 0 or -1 for vectors.",
+                    name_, codename_));
+        }
+
+        auto v = arg.vector();
+        double result = std::accumulate(v.begin(), v.end(), 0.);
+
+        if (keep_dims)
+        {
+            return primitive_argument_type{
+                blaze::DynamicVector<val_type>{result}};
+        }
+        return primitive_argument_type{result};
+    }
+
+    primitive_argument_type sum_operation::sum2d(arg_type&& arg,
+        hpx::util::optional<std::int64_t> axis, bool keep_dims) const
+    {
+        if (axis)
+        {
+            switch (axis.value())
+            {
+            case -2: HPX_FALLTHROUGH;
+            case 0:
+                return sum2d_axis0(std::move(arg));
+            case -1: HPX_FALLTHROUGH;
+            case 1:
+                return sum2d_axis1(std::move(arg));
+            default:
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "sum_operation::sum1d",
+                    execution_tree::generate_error_message(
+                        "the sum_operation primitive requires operand axis to be "
+                        "between -2 and 1 for matrices.",
+                        name_, codename_));
+            }
+        }
+        return sum2d_flat(std::move(arg), keep_dims);
+    }
+
+    primitive_argument_type sum_operation::sum2d_flat(
+        arg_type&& arg, bool keep_dims) const
+    {
+        auto m = arg.matrix();
+        double result = 0.;
+        for (std::size_t i = 0; i < m.columns(); ++i)
+        {
+            auto col = blaze::column(m, i);
+            result += std::accumulate(col.begin(), col.end(), 0.);
```

Comment:
The initial value should be `result`.

Reply:
I could write `result = std::accumulate(begin, end, result)` or `result += std::accumulate(begin, end, 0.)` as I did. What is wrong with it?